### PR TITLE
feat(shell-panel): add component tokens

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #7180

### Shell Panel

--calcite-shell-panel-background-color: Specifies the background color of the component.
--calcite-shell-panel-corner-radius: Specifies the corner radius of the component.
--calcite-shell-panel-detached-max-height: [Deprecated] Use the `heightScale` property instead. When `displayMode` is `float`, specifies the maximum height of the component.
--calcite-shell-panel-height: When `layout` is `horizontal`, or `layout` is `vertical` and `displayMode` is `float`, specifies the height of the component.
--calcite-shell-panel-max-height: When `layout` is `horizontal`, or `layout` is `vertical` and `displayMode` is `float`, specifies the maximum height of the component.
--calcite-shell-panel-max-width: Specifies the maximum width of the component.
--calcite-shell-panel-min-height: When `layout` is `horizontal`, or `layout` is `vertical` and `displayMode` is `float`, specifies the minimum height of the component.
--calcite-shell-panel-min-width: Specifies the minimum width of the component.
--calcite-shell-panel-shadow: Specifies the shadow of the component.
--calcite-shell-panel-width: Specifies the width of the component.
--calcite-shell-panel-z-index: Specifies the z-index value for the component.

### CODE REVIEW

File: packages/calcite-components/src/components/action/action.scss [hunk 1]
```diff
@@ -34,7 +34,7 @@
   }
 }
 
-button {
+.button {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/action/action.scss:37]: Are these changes intentional? They seem unrelated to `shell-panel`.

File: packages/calcite-components/src/components/shell-panel/shell-panel.scss [hunk 2]
```diff
@@ -179,97 +154,160 @@
...
+:host([layout="vertical"]) {
+  --calcite-internal-shell-panel-z-index: calc(theme("zIndex.default") + 1);
 
-:host([layout="vertical"]) .content {
-  inline-size: var(--calcite-shell-panel-width-internal);
-  max-inline-size: var(--calcite-shell-panel-max-width-internal);
-  min-inline-size: var(--calcite-shell-panel-min-width-internal);
-}
+  .content {
+    inline-size: var(--calcite-shell-panel-width, var(--calcite-internal-shell-panel-width));
```
Comment (jcfranco) [inline packages/calcite-components/src/components/shell-panel/shell-panel.scss:161]: Should `--calcite-shell-panel-width` be renamed to `--calcite-shell-panel-content-width`?

File: packages/calcite-components/src/components/shell-panel/shell-panel.scss [hunk 3]
```diff
@@ -12,7 +12,7 @@
  * @prop --calcite-shell-panel-min-height: When `layout` is `horizontal`, or `layout` is `vertical` and `displayMode` is `float`, specifies the minimum height of the component.
  * @prop --calcite-shell-panel-min-width: Specifies the minimum width of the component.
  * @prop --calcite-shell-panel-shadow: Specifies the shadow of the component.
- * @prop --calcite-shell-panel-width: Specifies the width of the component.
+ * @prop --calcite-shell-panel-content-width: Specifies the width of the component.
```
[the remark this answers is not in the document]
Comment (driskull) [reply packages/calcite-components/src/components/shell-panel/shell-panel.scss:15]: > It's also worth considering if we should just leave the initial name and apply this to the container itself (including slotted action bar), which I think it more in line with use case. A user can already control the width of slotted Panel, or other content, by setting width on that element.

That sounds reasonable to me.

**Review by jcfranco — APPROVED** [review]
